sdk/tests/utilities/test_token_refresh.py

```diff
@@ -186,3 +194,388 @@ def force_refresh(refresh_token):
 
             # Ensure that we only got an access token once
             self.assertEqual(1, identity_mock.call_count)
+
+    def test_retries_on_429_status_code_initial_access_token(self):
+        """
+        Ensures that in the event of a 429 HTTP Status Code being returned when communicating with an identity
+        provider, the request is retried.
+        """
+
+        refreshing_token = RefreshingToken(api_configuration=self.config)
+
+        with patch("requests.post", side_effect=[
+            # Return a 429 on the first attempt
+            MockApiResponse(
+                json_data={
+                    "error": "rate_limit",
+                    "error_description": "API rate limit exceeded."
+                },
+                status_code=429
+            ),
+            # Return a 200 on the second attempt
+            MockApiResponse(
+                json_data={
+                    "access_token": "mock_access_token",
+                    "refresh_token": "mock_refresh_token",
+                    "expires_in": 60
+                },
+                status_code=200
+            ),
+        ]):
+
+            # Ensure that we were able to get the token, if not retrying this would be impossible
+            self.assertEqual(f"{refreshing_token}", "mock_access_token")
```

Comment:
should assert on the call count of the mocks too (i.e. equals 1 for each of them)

Reply:
Added a call count assertion. As we make the API call with same inputs it is difficult to assert that there was one call for each returned status code. It would be impossible to get an access token if both calls weren't made so I feel that this is sufficient.